### PR TITLE
Log deprecation message on legacy ldap pool settings

### PR DESCRIPTION
**Pull request by peternied**
### Description
Log deprecation message on legacy ldap pool settings

Following up on comment https://github.com/opensearch-project/security/pull/2091#pullrequestreview-1113871093 from a contribution by @Martin-Kemp 

### Issues
- Resolves https://github.com/opensearch-project/security/issues/2059

### Testing
Manually tested - adding deprecation check involves migrating these tests to OpenSearchTestCase, far outside the scope of the existing security testing infrastructure, created https://github.com/opensearch-project/security/issues/2098

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java
Comment (cwperks) [inline src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: Would it make sense to move this function into Settings and pass the logger in so it can be re-used in other places?

Comment (peternied) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: Settings definitions themselves have a way to do this, see [Setting.java](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/Setting.java#L130).

As these legacy settings not registered into the schema when the plugin is registered we need a mechanism like I made here to support it.

Comment (cwperks) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: There are unregistered settings outside of LDAP as well. See https://github.com/opensearch-project/security/blob/main/src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java#L60 as an example. Does it make sense to move this in a utility class in `org.opensearch.security.support` in case we need to support this for other use cases in the future?

Comment (peternied) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: I would like to merge the code as is, in the future we could organize if we reuse it.  What do you think?

Comment (cwperks) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: I see another ticket in the backlog where this function could be applicable (it just so happens to also be related to LDAP): https://github.com/opensearch-project/security/issues/2059

Its functionally fine to have the method here, but I see an opportunity for re-use that we may miss in the future. @DarshitChanpura @cliu123 thoughts?

Comment (DarshitChanpura) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: I agree with the idea of making it more generic so it can be re-used as needed

**Review by cwperks — DISMISSED** [review]
Thanks @peternied! Can you link to the addition issue resolved by this PR (https://github.com/opensearch-project/security/issues/2059) in the PR description? 🎸

File: src/main/java/org/opensearch/security/setting/DeprecatedSettings.java
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/setting/DeprecatedSettings.java:11]: 👍

**Review by DarshitChanpura — COMMENTED** [review]
Ty Peter. I will approve once the CI passes.

**Review by DarshitChanpura — APPROVED** [review]

**Review by peternied — COMMENTED** [review]
Sorry for the momentarily disappearing changes, everything has been restored, all is well

**Review by cwperks — APPROVED** [review]

**Comment by cwperks** [comment]
@peternied Please resolve the unused import CI failure. This looks good to me. I will approve again once CI is passing.

**Review by cwperks — APPROVED** [review]

**Comment by peternied** [comment]
@opensearch-project/security Could I get another review?

File: src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java
Comment (DarshitChanpura) [inline src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: these method should be removed as it was moved out of this class

File: src/main/java/org/opensearch/security/setting/DeprecatedSettings.java
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/setting/DeprecatedSettings.java:10]: should this license be replaced with:
```
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

File: src/main/java/org/opensearch/security/setting/DeprecatedSettings.java
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/setting/DeprecatedSettings.java:31]: 👌🏽

File: src/test/java/org/opensearch/security/setting/DeprecatedSettingsTest.java
Comment (DarshitChanpura) [inline src/test/java/org/opensearch/security/setting/DeprecatedSettingsTest.java:10]: same comment as above for license

**Review by DarshitChanpura — COMMENTED** [review]
Ty @peternied for this! I have added a couple comments requesting small changes. I will approve once those are addressed.

File: src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java
Comment (peternied) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:138]: Oops! Removed

Comment (peternied) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:138]: Higher up in the file this method reference is statically imported, no change is needed for this to compile and use the proper method from DeprecatedSettings

File: src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java
Comment (peternied) [reply src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java:150]: Oops, removed!

**Review by DarshitChanpura — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Pull request merged**
